### ISSUE TITLE
Cant use ChromeCast

### ISSUE THREAD

**Issue by ivansteff** [google/shaka-player#2019]
Created a [project](https://github.com/ivansteff/ChromeCast-Problem-Shaka-player) using code as [an example](https://shaka-player-demo.appspot.com/docs/api/tutorial-ui.html). I do not understand why when i click on the cast button, an error arrives.
I want to make a caste from the browser on the computer to the TV( with Chromecast).

![image](https://user-images.githubusercontent.com/31563237/60645520-8dc3e800-9e41-11e9-843b-bcf548257f3b.png)

**Comment by joeyparrish** [comment]
In [your code](https://github.com/ivansteff/ChromeCast-Problem-Shaka-player/blob/ec37b058649db96bbef9c6a2cee3c434c0adb990/index.html#L5-L6), I see two different copies of Shaka Player included:

```html
<script
  src="https://cdnjs.cloudflare.com/ajax/libs/shaka-player/2.5.2/shaka-player.compiled.js"
  crossorigin="anonymous"></script>
<script
  src="https://cdnjs.cloudflare.com/ajax/libs/shaka-player/2.5.2/shaka-player.ui.js"
  crossorigin="anonymous"></script>
```

You don't need both of those, and they may conflict.  Please try removing "shaka-player.compiled.js", and just use the UI-enabled version of the library.

**Comment by ivansteff** [comment]
@joeyparrish[ I removed](https://github.com/ivansteff/ChromeCast-Problem-Shaka-player) the compiled version, left only ui. However, the error still repeats.

**Comment by joeyparrish** [comment]
I'm at a loss.  There are so many things wrong here.

Here's what I've seen by running the code in your repo:

1. The styling is wrong on the cast button, so it looks way out of place
2. Clicking outside of the overflow menu causes all overflow submenus to show up, and then they can't be dismissed
3. The cast button shows up even if there are no Chromecast devices on the network
4. Clicking the cast button throws a UI error, but it's not an error, it's a MouseEvent

**Comment by joeyparrish** [comment]
I found the problem.  You are overriding the `.shaka-hidden` style, which causes the button to show up when it shouldn't:

https://github.com/ivansteff/ChromeCast-Problem-Shaka-player/blob/f38e1d33f2a0b0ef08caf005669b400b4a71eb03/index.html#L9-L13

```html
    <style>
    .shaka-hidden, button.shaka-hidden{
        display: block;
    }
    </style>
```

In short, please don't do that.  :-)

The reason the button is hidden is that there are no cast devices available on your network or your browser doesn't support casting (such as Firefox, Edge, or Safari).  So when you click it, it doesn't work, naturally.

Forcing hidden things to be shown is also the cause of the other weird style issues above, such as the overflow submenus being shown when the menus should be dismissed.

**Comment by ivansteff** [comment]
@joeyparrish Ok I understood. But my network has a [xaomi tv](https://www.mi.com/in/mi-led-tv-4a-pro-43/) with chromecast function. i can cast from [cast-videos](https://codelabs.developers.google.com/codelabs/cast-videos-chrome/#0),
and can cast a page from Google Chrome.

**Comment by joeyparrish** [comment]
Ah, I see.  It appears that you are missing the Cast SDK:

```html
    <script defer src="https://www.gstatic.com/cv/js/sender/v1/cast_sender.js"></script>
```

Try adding that and see if you can cast.

**Comment by ivansteff** [comment]
Thanks for the reply, I apologize for my delay in replying.
I [added](https://github.com/ivansteff/ChromeCast-Problem-Shaka-player/commit/ee9c4ff845220bd99d0b81de48a8b972020ccae6#diff-eacf331f0ffc35d4b482f1d15a887d3bR16) sdk, nothing has changed.
But I can broadcast the page on TV.
![image](https://user-images.githubusercontent.com/31563237/61052133-6ea1f900-a3f3-11e9-8b28-a3e0223bde0a.png)

**Comment by joeyparrish** [comment]
Okay, now that we've fixed your duplicate libraries, extra styles, and missing cast SDK, it should be working.  But it isn't.  In fact, it isn't working in the demo app either, in v2.5.1, v2.5.2, or v2.5.3.

At this point, we definitely have a bug, and we'll dig into what that is.  Thanks!

**Comment by joeyparrish** [comment]
Oh, but the bug is not present in the nightly build, so we must have already fixed it in the master branch.  I'll compare master to v2.5.x and see if I can spot the difference.

**Comment by ivansteff** [comment]
@joeyparrish  Okay, Thanks!
I will wait.

**Comment by TheModMaker** [comment]
It looks like it was fixed in 3e3e940 but wasn't cherry-picked for v2.5.2.  I've cherry-picked it to the v2.5.x branch so it will appear in the next release.
